lib/OpenQA/JobSettings.pm

```diff
@@ -52,29 +52,32 @@ sub generate_settings ($params) {
 }
 
 # replace %NAME% with $settings{NAME} (but not %%NAME%%)
-sub expand_placeholders ($settings) {
+sub expand_placeholders ($settings, $on_web_ui = 1) {
     for my $value (values %$settings) {
         next unless defined $value;
         my %visited_placeholders;
-        eval { $value =~ s/(%+)(\w+)(%+)/_expand_placeholder($settings, $2, $1, $3, \%visited_placeholders)/eg };
+        eval {
+            $value =~ s/(%+)(\w+)(%+)/_expand_placeholder($settings, $2, $1, $3, \%visited_placeholders, $on_web_ui)/eg;
+        };
         return "Error: $@" if $@;
     }
     return undef;
 }
 
-sub _expand_placeholder ($settings, $key, $start, $end, $visited_placeholders_in_parent_scope) {
-    return '' unless defined $settings->{$key};
+sub _expand_placeholder ($settings, $key, $start, $end, $visited_placeholders_in_parent_scope, $on_web_ui) {
+    # handle %CASEDIR% only on web UI; on the worker it is handled by _engine_workit_step_2 separately
```

Comment:
This should be added in the documentation above

Reply:
No, it is already documented in a more fitting section.